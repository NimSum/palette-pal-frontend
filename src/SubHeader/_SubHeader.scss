.SubHeader {
  height: 62px;
  background: white;
  border-bottom: 1px solid #E3E3E3;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
}

.page-title {
  font-size: 22px;
  color: #28554D;
}

.subheader-left {
  display: flex;
}

.subheader-right {
  display: flex;
}

.subheader-btn {
  margin: 0 35px;
  padding: 0 10px;
  border-radius: 5px;
  color: $btn-color;
  border-color: $btn-color;

  .fa, .fas {
    margin-right: 5px;
  }

  &:hover {
    color: $btn-hover;
    border-color: $btn-hover;
  }
}

.dropdown-label {
  font-size: 16px;
  color: #4a4a4a;
  margin-left: 20px;
}

.dropdown-input {
  min-width: 100px;
  margin: 0 8px;
}