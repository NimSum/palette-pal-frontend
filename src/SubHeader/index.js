import React, { Component } from 'react';

class SubHeader extends Component {
  constructor(props) {
    super(props);

    this.state = {
      format: '',
      mode: '',
      filter: ''
    }
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
    this.props.setOption({ [target.name]: target.value });
  }
    
  render() {
    const btnName = this.props.title === "Pick New Palette" ? "Generate New Palette" : "Create New Project";

    const projectOptions = this.props.data.map(project => <option value={project.id} key={project.id} >{project.name}</option>)

    const filters = this.props.title === "Pick New Palette" ?  <>
        <p className="dropdown-label">Format:</p><select className="dropdown-input" type="text" name="format"/>
        <p className="dropdown-label">Mode:</p><select className="dropdown-input" type="text" name="mode"/>
    </>
      : <>
        <p className="dropdown-label">Filter:</p>
        <select
          className="dropdown-input"
          type="text"
          name="filter"
          value={this.state.filter}
          onChange={this.handleChange}
        >
          <option value=''>All Projects</option>
          {projectOptions}
        </select>
    </>

    return (
      <section className="SubHeader">
        <div className="subheader-left">
          <h2 className="page-title">{this.props.title}</h2>
          <button className="subheader-btn" onClick={this.props.handleClick}><i className="fas fa-play" aria-hidden="true"></i>{btnName}</button>
        </div>
        <div className="subheader-right">
          {filters}
        </div>
      </section>
    );
  }
}

export default SubHeader;