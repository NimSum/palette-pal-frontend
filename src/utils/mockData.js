const mockData = {
  mockDetailedProjects: [
    {
      "project_name": "Project ONE",
      "project_id": 1,
      "palette_name": "Palette ONE",
      "palette_id": 1,
      "color_1": "#000000",
      "color_2": "#000000",
      "color_3": "#319547",
      "color_4": "#000000",
      "color_5": "#834267"
    },
    {
      "project_name": "Project TWO",
      "project_id": 2,
      "palette_name": "Palette TWO",
      "palette_id": 2,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267"
    },
    {
      "project_name": "Project ONE",
      "project_id": 2,
      "palette_name": "Palette THREE",
      "palette_id": 3,
      "color_1": "#123456",
      "color_2": "#654321",
      "color_3": "#245216",
      "color_4": "#784895",
      "color_5": "#235686"
    }
  ],
  mockProjects: [
    {
      "id": 1,
      "project_name": "Project ONE",
      "created_at": "2019-07-05T21:39:51.436Z",
      "updated_at": "2019-07-05T21:39:51.436Z"
    },
    {
      "id": 2,
      "project_name": "Project TWO",
      "created_at": "2019-07-05T21:40:51.436Z",
      "updated_at": "2019-07-05T21:40:51.436Z"
    }
  ],
  mockPalettes: [
    {
      "id": 1,
      "palette_name": "Test ONE",
      "project_id": 1,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267",
      "created_at": "2019-07-05T21:39:57.227Z",
      "updated_at": "2019-07-05T21:39:57.227Z"
    },
    {
        "id": 2,
        "palette_name": "Test TWO",
        "project_id": 2,
        "color_1": "#000000",
        "color_2": "#000000",
        "color_3": "#319547",
        "color_4": "#000000",
        "color_5": "#834267",
        "created_at": "2019-07-06T01:24:45.881Z",
        "updated_at": "2019-07-06T01:24:45.881Z"
    }
  ]
}

export default mockData;