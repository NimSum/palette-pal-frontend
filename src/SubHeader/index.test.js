import React from 'react';
import SubHeader from './index';
import { shallow } from 'enzyme';
import mockData from '../utils/mockData';

describe('SubHeader', () => {
  let wrapper;
  let instance;
  const mockSetOption = jest.fn();
  const mockHandleClick = jest.fn();

  beforeEach(() => {
    wrapper = shallow(< SubHeader
      data={[mockData.mockCleanedProject]}
      setOption={mockSetOption}
      handleClick={mockHandleClick}
      title="Pick New Palette"
    />);

    instance = wrapper.instance();
    jest.spyOn(instance, 'handleChange');
  })
  
  it('should match component SubHeader snapshot when the title is Pick New Palette', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match component SubHeader snapshot when the title is My Projects', () => {
    wrapper = shallow(< SubHeader
      data={[mockData.mockCleanedProject]}
      setOption={mockSetOption}
      title="My Projects"
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have the expected default state', () => {
    expect(wrapper.state()).toEqual({
      format: 'hex',
      mode: 'random',
      project: ''
    })
  })

  it('should update state and call setOption when handleChange is called', () => {
    const mockEvent = { target: { name: 'mode', value: 'test value' } }

    expect(wrapper.state('mode')).toEqual('random');
    
    instance.handleChange(mockEvent);

    expect(wrapper.state('mode')).toEqual('test value');
    expect(mockSetOption).toHaveBeenCalledWith({ mode: 'test value'})
  })

  it('should invoke handleChange when the mode dropdown changes', () => {
    const mockEvent = { target: { name: 'mode', value: 'test value' } }

    expect(wrapper.state('mode')).toEqual('random');

    wrapper.find('.mode-dropdown').simulate('change', mockEvent)

    expect(wrapper.state('mode')).toEqual('test value');
  })

  it('should invoke handleChange when the project dropdown changes', () => {
    wrapper = shallow(< SubHeader
      data={[mockData.mockCleanedProject]}
      setOption={mockSetOption}
      title="My Projects"
    />);
    
    const mockEvent = { target: { name: 'project', value: 'test proj value' } }

    expect(wrapper.state('project')).toEqual('');

    wrapper.find('.project-dropdown').simulate('change', mockEvent)

    expect(wrapper.state('project')).toEqual('test proj value');
  })

  it('should invoke handleChange when the format dropdown changes', () => {
    const mockEvent = { target: { name: 'format', value: 'test format value' } }

    expect(wrapper.state('format')).toEqual('hex');

    wrapper.find('.format-dropdown').simulate('change', mockEvent)

    expect(wrapper.state('format')).toEqual('test format value');
  })

  it('should call handleClick from props when the subheader button is clicked', () => {
    wrapper.find('.subheader-btn').simulate('click');

    expect(mockHandleClick).toHaveBeenCalled();
  })
})