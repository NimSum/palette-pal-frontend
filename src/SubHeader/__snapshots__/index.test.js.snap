// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader shoud match component SubHeader snapshot 1`] = `ShallowWrapper {}`;
